test/insertion.bats

```diff
@@ -1,38 +1,111 @@
 load test_helper
 
-@test "insertion target" {
-    result="$(p -a Alamofire -t Altel -f $suts/podfile-altel -d)"
+setup() {
+  init
+}
+
+teardown() {
+  rm -rf $suts/temp
+}
+
+@test "insertion into target" {
+    result="$(p -a Alamofire -t Altel -f $suts/temp -d)"
     text="use_frameworks!
 platform :ios, '10.0'
 inhibit_all_warnings!
 
 target 'Altel' do
 pod 'Alamofire'
   pod 'R.swift'
-  pod 'Sourcery'
   pod 'SwiftFormat/CLI'
-  pod 'SwiftLint'
+end
+
+def insert_pods
+  pod 'Typhoon', '~>4.0’
+  # Analytics
 end"
     [ "$result" == "$text" ]
 }
 
-@test "insertion def" {
-   result="$(p -a Alamofire -g insert_pods -f $suts/podfile-gleam -d)"
-   text="def insert_pods
+@test "insertion into: 1) def group" {
+   result="$(p -a Alamofire -g insert_pods -f $suts/temp -d)"
+   text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+
+target 'Altel' do
+  pod 'R.swift'
+  pod 'SwiftFormat/CLI'
+end
+
+def insert_pods
 pod 'Alamofire'
   pod 'Typhoon', '~>4.0’
-  pod 'NVActivityIndicatorView', '~>4.4'
-  pod 'SnapKit', '~>4.2'
-  pod 'RxSwift', '~>4.2'
-  pod 'DTCollectionViewManager', '~> 6.4'
-  pod 'HCSStarRatingView', '~> 1.5'
-  pod 'Moya', '~> 11.0'
+  # Analytics
+end"
+    [ "$result" == "$text" ]
+}
+
+@test "insertion into: 2) comment group" {
+   result="$(p -a Alamofire -g Analytics -f $suts/temp -d)"
+   text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+
+target 'Altel' do
+  pod 'R.swift'
+  pod 'SwiftFormat/CLI'
 end
 
-target 'Gleam' do
-  use_frameworks!
-  # Pods for Gleam
-  insert_pods
+def insert_pods
+  pod 'Typhoon', '~>4.0’
+  # Analytics
+  pod 'Alamofire'
 end"
     [ "$result" == "$text" ]
 }
+
+@test "insertion into root" {
+  result="$(p -a Alamofire -p "root" -f $suts/temp -d)"
+  text="use_frameworks!
+platform :ios, '10.0'
+inhibit_all_warnings!
+pod 'Alamofire'
+
+target 'Altel' do
+  pod 'R.swift'
+  pod 'SwiftFormat/CLI'
+end
+
+def insert_pods
+  pod 'Typhoon', '~>4.0’
+  # Analytics
+end"
+[ "$result" == "$text" ]
+}
+
+@test "skipping reserved word" {
```

Comment:
I think we can move these testcases to separate file and split them one by one.

Reply:
Also after moving to new file we need to add `test` command inside `Makefile` which runs all tests inside `test/` folder.

Reply:
We think that separating tests into different files is not good idea. Isn't it enough to have one file and run it inside Makefile?

Reply:
Separation is generally a good idea when it comes to file organization. No one loves bloated stuff :)
Usually you can most of the testcases in one file, but in most of the testing frameworks there is a way to group different testing scenarios for better readability of the test code as well as better readability of the output.

Here is the example usage of Jest (JS testing framework):

```js
describe('testing arithmetical functions', () => {
  it('should add numbers properly', () => {
     expect(2+2).toBeEqual(4)
   });
   it('should multiply numbers properly', () => {
     expect(3*2).toBeEqual(6)
   });
});

describe('testing some other stuff', () => {
 // ...
});

```

Here we can clearly separate two different scenarios "testing arithmetic functions" and "some other stuff". Output of above testcases would also be properly formatted in the console. As well as it provides better readability for the developers. But due to the fact that we are limited with what `bats` provides us here - we need to opt for just moving grouped testcases to separate files :)